cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/PredefinedAbstainDRep.hs

```diff
@@ -0,0 +1,362 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.Testnet.Test.LedgerEvents.Gov.PredefinedAbstainDRep
+  ( hprop_check_predefined_abstain_drep
+  ) where
+
+import           Cardano.Api as Api
+import           Cardano.Api.Eon.ShelleyBasedEra (ShelleyLedgerEra)
+import           Cardano.Api.Error (displayError)
+
+import           Cardano.Ledger.Conway.Core (ppNOptL)
+import           Cardano.Ledger.Conway.Governance (ConwayGovState, cgsCurPParamsL)
+import           Cardano.Ledger.Core (EraPParams)
+import           Cardano.Ledger.Shelley.LedgerState (epochStateGovStateL, nesEpochStateL)
+import           Cardano.Testnet
+
+import           Prelude
+
+import           Control.Monad (void)
+import           Control.Monad.Catch (MonadCatch)
+import           Data.String (fromString)
+import qualified Data.Text as Text
+import           Data.Word (Word32)
+import           GHC.Stack (HasCallStack, callStack)
+import           Lens.Micro ((^.))
+import           System.FilePath ((</>))
+
+import           Testnet.Components.DReps (createCertificatePublicationTxBody, createVotingTxBody,
+                   generateVoteFiles, retrieveTransactionId, signTx, submitTx)
+import           Testnet.Components.Query (EpochStateView, findLargestUtxoForPaymentKey,
+                   getCurrentEpochNo, getEpochStateView, getGovState, getMinDRepDeposit,
+                   waitAndCheckNewEpochState)
+import           Testnet.Defaults (defaultDRepKeyPair, defaultDelegatorStakeKeyPair)
+import qualified Testnet.Process.Cli as P
+import qualified Testnet.Process.Run as H
+import qualified Testnet.Property.Utils as H
+import           Testnet.Runtime
+
+import           Hedgehog
+import qualified Hedgehog.Extras as H
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+
+-- | This test creates a default testnet with three DReps and one stake holder delegated to each.
+-- We then do a proposal for an arbitrary parameter change (in this case to the
+-- @desiredNumberOfPools@ parameter) to check that it fails, when the first DRep votes "yes" and the
+-- last two vote "no". Later we chack that if we change the stake holders under the DReps that vote
+-- "no" to delegate to the automate "always abstain" DRep, the same kind of proposal passes.
+--
+-- This test is meant to ensure that delegating to "always abstain" has the desired effect of
+-- counting as abstaining for the stake delegated.
+--
+-- Execute me with:
+-- @DISABLE_RETRIES=1 cabal test cardano-testnet-test --test-options '-p "/Predefined Abstain DRep/"'@
+hprop_check_predefined_abstain_drep :: Property
+hprop_check_predefined_abstain_drep = H.integrationWorkspace "test-activity" $ \tempAbsBasePath' -> do
```

Comment:
```suggestion
hprop_check_predefined_abstain_drep = H.integrationWorkspace "test-activity" $ \tempAbsBasePath' -> runWithDefaultWatchdog_ $ do
```
We need this otherwise the test case may hang.

Reply:
Fixed here: https://github.com/IntersectMBO/cardano-node/pull/5815/commits/846e70aec7d0fb8c8aecb954a883f4f1f4c9d489